plugins-extra/query_enhancements/public/query_assist/components/query_assist_bar.tsx

```diff
@@ -0,0 +1,95 @@
+import { EuiFlexGroup, EuiFlexItem, EuiForm, EuiFormRow } from '@elastic/eui';
+import React, { SyntheticEvent, useEffect, useMemo, useRef, useState } from 'react';
+import { IDataPluginServices, PersistedLog } from '../../../../../src/plugins/data/public';
+import { SearchBarExtensionDependencies } from '../../../../../src/plugins/data/public/ui/search_bar_extensions/search_bar_extension';
+import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { getStorage } from '../../services';
+import { useGenerateQuery } from '../hooks';
+import { getPersistedLog, ProhibitedQueryError } from '../utils';
+import { QueryAssistCallOut, QueryAssistCallOutType } from './call_outs';
+import { QueryAssistInput } from './query_assist_input';
+import { QueryAssistSubmitButton } from './submit_button';
+
+interface QueryAssistInputProps {
+  dependencies: SearchBarExtensionDependencies;
+}
+
+export const QueryAssistBar: React.FC<QueryAssistInputProps> = (props) => {
+  const { services } = useOpenSearchDashboards<IDataPluginServices>();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const storage = getStorage();
+  const persistedLog: PersistedLog = useMemo(
+    () => getPersistedLog(services.uiSettings, storage, 'query-assist'),
+    [services.uiSettings, storage]
+  );
+  const { generateQuery, loading } = useGenerateQuery();
+  const [callOutType, setCallOutType] = useState<QueryAssistCallOutType>();
+  const dismissCallout = () => setCallOutType(undefined);
+  const mounted = useRef(false);
+  const selectedIndex = props.dependencies.indexPatterns?.at(0)?.title;
+  const previousQuestionRef = useRef<string>();
+
+  useEffect(() => {
+    mounted.current = true;
+    return () => {
+      mounted.current = false;
+    };
+  }, []);
+
+  const onSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    if (!inputRef.current?.value) {
+      setCallOutType('empty_query');
+      return;
+    }
+    if (!selectedIndex) {
+      setCallOutType('empty_index');
+      return;
+    }
+    dismissCallout();
+    previousQuestionRef.current = inputRef.current.value;
+    persistedLog.add(inputRef.current.value);
+    const params = {
+      question: inputRef.current.value,
+      index: selectedIndex,
+      language: 'PPL',
```

Comment:
Future proofing, could we make this more generic even if it only works for PPL right now?

The last query in the query service should have the language attached to it assuming that is the language. And even more straight forward approach and preferred could be pulling from localstorage:
![Screenshot 2024-06-04 at 14-12-41 (11) Discover UI Updates - Quip](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/12060704/1294b5dc-6e0e-42a3-bdc8-eff66e4b2440)

which you can access from the ui service in the data plugin: 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/ui_service.ts#L65

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/settings/settings.ts#L28

Reply:
good point i missed moving this to a variable

query assist can run before before any query runs, would localStorage be set after changing language or executing query? I tested on my side but didn't see this key in localStorage

i think i can still pass in language when creating the query assist search bar extension, since currently they are language specific